From 980a95ef519c6f014e22d2eae31c735b31555427 Mon Sep 17 00:00:00 2001
From: Rohan Puri <rohan_puri@xyratex.com>
Date: Fri, 25 Oct 2013 14:27:19 +0530
Subject: [PATCH 111/121] Changes summary : - 1. Added 2 @todo   - Stats
 records fop's rpc item deadline value needs to be performance tuned.   -
 Currently in m0d we connect with stats service in m0_cs_start(), should be   
  moved to addb pfom, tried but facing some initialization ordering issues.

These @todos are part of round 1 review of addb-monitoring-infrastructure
at link http://reviewboard.clusterstor.com/r/1544/
---
 addb/addb_monitor.c | 4 ++++
 mero/setup.c        | 5 +++++
 2 files changed, 9 insertions(+)

diff --git a/addb/addb_monitor.c b/addb/addb_monitor.c
index cf5ccaa..41c99f7 100644
--- a/addb/addb_monitor.c
+++ b/addb/addb_monitor.c
@@ -445,6 +445,10 @@ static int addb_monitor_stats_fop_send(struct m0_stats_update_fop *fop_data,
 
 	item                     = &stats_update_fop->f_item;
 	item->ri_prio            = M0_RPC_ITEM_PRIO_MIN;
+	/**
+	 * @todo: Need to check what should be optimal value for
+	 * item deadline. Should be done during performance optimization.
+	 */
 	item->ri_deadline        = 0;
 
 	rc = m0_rpc_oneway_item_post(conn, item);
diff --git a/mero/setup.c b/mero/setup.c
index 4dd610b..e8affbd 100644
--- a/mero/setup.c
+++ b/mero/setup.c
@@ -2214,6 +2214,11 @@ int m0_cs_start(struct m0_mero *cctx)
 	M0_ENTRY();
 	M0_PRE(cctx != NULL);
 	rc = cs_services_init(cctx);
+	/**
+	 * @todo: stats connection initialization should be done
+	 * in addb pfom, currently not being done due to some initialization
+	 * ordering issues.
+	 */
 	if (cctx->cc_stats_svc_epx.ex_endpoint != NULL)
 		cs_addb_monitor_stats_svc_conn_init(cctx);
 
-- 
1.8.3.2

